lxd/auth/oidc/oidc.go

```diff
@@ -308,70 +308,107 @@ func (o *Verifier) setAccessTokenVerifier() error {
 	return nil
 }
 
-// getCookies gets the ID and refresh tokens from the request cookies.
-func (*Verifier) getCookies(r *http.Request) (idToken string, refreshToken string, err error) {
+// getCookies gets the sessionID, identity and refresh tokens from the request cookies and decrypts them.
+func (o *Verifier) getCookies(r *http.Request) (sessionID uuid.UUID, idToken string, refreshToken string, err error) {
```

Comment:
Lets return a pointer to a uuid.UUID here to allow returning nil when erroring rather than having to initialise and return empty structs.

Reply:
Also done.